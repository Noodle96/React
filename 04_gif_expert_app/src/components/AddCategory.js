//rafc
import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {setInputValue(e.target.value)}
    const handleSubmit = (e) =>  {
        e.preventDefault();
        // console.log('hecho submit');
        if(inputValue.trim().length > 1){
            setCategories((cats) => [...cats, inputValue]);
            setInputValue('');
        }
    }
    return (
        <form onSubmit={ (e) => handleSubmit(e)}>
            {/* <h1>{inputValue}</h1> */}
            <input 
                type='text'
                value={inputValue}
                onChange={(e) => handleInputChange(e)}
            />
        </form>
    );
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}