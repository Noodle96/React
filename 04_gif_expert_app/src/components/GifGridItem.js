import React,{Fragment} from 'react'

export const GifGridItem = ({id,title,url,type,username}) => {
    // console.log({id,title,url});
    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>

        // <Fragment>
        //     <div className="wrapper">
        //         <div className="product-img">
        //         <img src={url} alt={title} />
        //         </div>
        //         <div className="product-info">
        //             <div className="product-text">
        //                 <h1 >{title}</h1>
        //                 <h2>{type}</h2>
        //                 <p>{username}  </p>

        //             </div>
        //             <div className="product-price-btn">
        //                  <p><span>Free</span></p>
        //                  <button type="button">Download</button>
        //              </div>
        //         </div>
        //     </div>
        // </Fragment>
    )
}



