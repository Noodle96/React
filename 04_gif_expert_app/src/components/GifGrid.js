import React, { useState,useEffect,Fragment } from 'react'
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    // const [images, setImages] = useState([]);
    const {data,loading} = useFetchGifs();
    
    // useEffect(() => {
    //     getGifs(category)
    //         .then( imgs => setImages(imgs));
    // },[category]);

    // const [contador, setcontador] = useState(0);
    
    

    // getGifs();

    return (
        <Fragment>
            <h3>{category}</h3>
            {loading ? 'Cargando...': 'Data Cargada'}
            {/* <div className='card-grid'>
                
                {
                    images.map(img => (
                        // <li key={img.id}>{img.title}</li>
                        <GifGridItem 
                            key={img.id}
                            // img={img}
                            {...img}
                        />
                    ))
                }
            </div> */}
        </Fragment>
    )
}
