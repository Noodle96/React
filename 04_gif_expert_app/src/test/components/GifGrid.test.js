import React from "react";
import {shallow} from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
describe('Pruebas en el componente GifGrid', () => {
    const category = 'template category';
    let wrapper = shallow(<GifGrid category={category}/>);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<GifGrid category={category}/>);
    });
    
    test('El componente <GifGrid /> deberia de mostrarse correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    })    
});