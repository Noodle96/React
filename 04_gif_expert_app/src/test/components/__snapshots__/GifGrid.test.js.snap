// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid El componente <GifGrid /> deberia de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__backInDown"
  >
    template category
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
